File: dotnet/PersonasABM.Application/Interfaces/IPersonaService.cs
using PersonasABM.Application.DTOs;

namespace PersonasABM.Application.Interfaces;

public interface IPersonaService
{
    Task<IEnumerable<PersonaDto>> GetAllAsync();
    Task<PersonaDto?> GetByIdAsync(int id);
    Task<PersonaDto> CreateAsync(CreatePersonaDto createPersonaDto);
    Task<PersonaDto> UpdateAsync(int id, UpdatePersonaDto updatePersonaDto);
    Task DeleteAsync(int id);
    Task<IEnumerable<PersonaDto>> SearchAsync(PersonaSearchDto searchDto);
    Task<bool> ValidateIdentificacionAsync(string identificacion, int? excludeId = null);
}

File: nodejs/src/services/IPersonaService.ts
import { PersonaDto, CreatePersonaDto, UpdatePersonaDto, PersonaSearchDto } from '../dto/PersonaDto';

export interface IPersonaService {
  findAll(): Promise<PersonaDto[]>;
  findById(id: number): Promise<PersonaDto | null>;
  create(createPersonaDto: CreatePersonaDto): Promise<PersonaDto>;
  update(id: number, updatePersonaDto: UpdatePersonaDto): Promise<PersonaDto>;
  delete(id: number): Promise<void>;
  search(searchDto: PersonaSearchDto): Promise<PersonaDto[]>;
  validateIdentificacion(identificacion: string, excludeId?: number): Promise<boolean>;
}
